tools/wake-format/README.md

```diff
@@ -0,0 +1,481 @@
+# wake-format
+
+`wake-format` is a tool for automatically applying the wake style guide to wake source files. It is intentionally opinionated to create a singular correct formatting standard.
+
+# Usage
+
+The standard usage of `wake-format` is to format a set of wake files in place.
+
+`wake-format -i source.wake source2.wake source3.wake`
+
+`wake-format` supports other flows for various use cases. The `-n`/`--dry-run` flag is ideal for CI/CD to check if the source needs to be formatted. It exits with a non-zero status code when a file would be changed by `wake-format`
+
+Running `wake-format` without `-i` will emit the file to `stdout` instead of overwritting it.
+
+# Documentation
+
+The information below should be helpful to contribute to `wake-format`
+
+## Building
+
+`wake-format` is built in the same manner as `wake`. Running `make` or `./bin/wake build default` from within the wake dir will produce the `wake-format` binary.
+
+The locally build `wake-format` can be found at `./bin/wake-format.native-cpp14-release`
+
+## wake-format Fluent API/DSL
+
+`wake-format` relies heavily on a custom fluent API/DSL to generate the formatting rules. Usage of the API is documented below. See [here](TODO) for design decisions, invariants, and engineering tradeoffs considered.
+
+### Usage
+
+All formatting rules start by creating a basic formatting object using `fmt()`. From there functions are added to describe the expectations of the current [CST node](TODO) and to return the appropiate text.
```

Comment:
> current CST node

It may be helpful to spell out the acronym as individual words the first time it's used

Reply:
It should be noted that this documentation is largely intended for wake devs at this time, we'll add user documentation once we have a user interface :)

Reply:
Yep I can see that, mostly I myself had forgotten what CST stood for